### PR TITLE
samples: create a Pub/Sub export subscription

### PR DESCRIPTION
**Pull request by tmdiep**
Added a sample for creating an export subscription to a Pub/Sub topic and test.

### CODE REVIEW

File: samples/snippets/create_lite_pubsub_export_subscription_example.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    cloud_region,
+    zone_id,
+    topic_id,
+    subscription_id,
+    pubsub_topic_id,
+    regional,
+    target_location,
+):
+    # [START pubsublite_create_pubsub_export_subscription]
+    from google.api_core.exceptions import AlreadyExists
```
Comment (kurtisvg) [inline samples/snippets/create_lite_pubsub_export_subscription_example.py:36]: Imports should be outside the main function: https://googlecloudplatform.github.io/samples-style-guide/#imports

File: samples/snippets/create_lite_pubsub_export_subscription_example.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    client = AdminClient(cloud_region)
+    try:
+        response = client.create_subscription(subscription, target_location)
+        print(f"{response.name} created successfully.")
+    except AlreadyExists:
+        print(f"{subscription_path} already exists.")
+    # [END pubsublite_create_pubsub_export_subscription]
+
+
+if __name__ == "__main__":
```
Comment (kurtisvg) [inline samples/snippets/create_lite_pubsub_export_subscription_example.py:97]: We should avoid CLIs: https://googlecloudplatform.github.io/samples-style-guide/#no-cli

File: samples/snippets/create_lite_pubsub_export_subscription_example.py [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+
+    # TODO(developer):
+    # project_number = 1122334455
+    # cloud_region = "us-central1"
+    # zone_id = "a"
+    # topic_id = "your-topic-id"
+    # subscription_id = "your-subscription-id"
+    # pubsub_topic_id = "destination-topic-id"
+    # regional = True
+    # target_location = BacklogLocation.BEGINNING
```
Comment (kurtisvg) [inline samples/snippets/create_lite_pubsub_export_subscription_example.py:54]: We shouldn't include commented out code, and should use minimal arguments: https://googlecloudplatform.github.io/samples-style-guide/#minimal-arguments

For suggesting default values for users, python has two options:
1. Default arguments:
```python
def example_snippet(project_number="1122334455", cloud_region="us-central1"):
```

2. Use an "easy-run" wrapper function:
```python
def run_snippet():
    project_number = "1122334455"
    cloud_region = "us-central1"
   example_snippet(project_number, region)   

def example_snippet(project_number, region):
```

File: samples/snippets/create_lite_pubsub_export_subscription_example.py [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+            # - `ACTIVE`: enable message processing.
+            # - `PAUSED`: suspend message processing.
+            desired_state=ExportConfig.State.ACTIVE,
+            pubsub_config=ExportConfig.PubSubConfig(
+                topic=destination_topic_path,
+            ),
+        ),
+    )
+
+    client = AdminClient(cloud_region)
```
Comment (kurtisvg) [inline samples/snippets/create_lite_pubsub_export_subscription_example.py:88]: Please add a comment about initializing the client: https://googlecloudplatform.github.io/samples-style-guide/#clients

File: samples/snippets/quickstart_test.py [hunk 5]
```diff
@@ -202,6 +206,27 @@ def regional_subscription(client, regional_topic):
...
+
+
+@pytest.fixture(scope="module")
+def pubsub_topic(publisher_client):
+    """Creates a Pub/Sub topic resource"""
+    destination_topic_path = publisher_client.topic_path(PROJECT_NUMBER, PUBSUB_TOPIC_ID)
+    try:
+        publisher_client.create_topic(request={"name": destination_topic_path})
+    except AlreadyExists:
+        None
```
Comment (kurtisvg) [inline samples/snippets/quickstart_test.py:221]: Please log the error so it's visible for debugging

**Review by kurtisvg — CHANGES_REQUESTED** [review]

File: samples/snippets/create_lite_pubsub_export_subscription_example.py [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This application demonstrates how to create a subscription that writes
+messages to a Pub/Sub topic. For more information, see documentation at
+https://cloud.google.com/pubsub/lite/docs/export-subscriptions.
```
Comment (anguillanneuf) [inline samples/snippets/create_lite_pubsub_export_subscription_example.py:19]: Do we know when this page will be up?

File: samples/snippets/create_lite_pubsub_export_subscription_example.py [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+    project_number,
+    cloud_region,
+    zone_id,
+    topic_id,
+    subscription_id,
+    pubsub_topic_id,
+    regional,
+    target_location,
+):
+    # [START pubsublite_create_pubsub_export_subscription]
```
Comment (anguillanneuf) [inline samples/snippets/create_lite_pubsub_export_subscription_example.py:35]: Have you coordinated with Moi to create this region tag first?

**Review by anguillanneuf — CHANGES_REQUESTED** [review]

File: samples/snippets/create_lite_pubsub_export_subscription_example.py (hunk 7 above)
Comment (tmdiep) [reply samples/snippets/create_lite_pubsub_export_subscription_example.py:35]: Moi said the region tag will be created soon (b/227170276).

File: samples/snippets/create_lite_pubsub_export_subscription_example.py (hunk 6 above)
Comment (tmdiep) [reply samples/snippets/create_lite_pubsub_export_subscription_example.py:19]: The page will be up in mid December. I'll update this link if the page name changes.

File: samples/snippets/create_lite_pubsub_export_subscription_example.py (hunk 2 above)
Comment (tmdiep) [reply samples/snippets/create_lite_pubsub_export_subscription_example.py]: I've removed the CLI

File: samples/snippets/create_lite_pubsub_export_subscription_example.py (hunk 3 above)
Comment (tmdiep) [reply samples/snippets/create_lite_pubsub_export_subscription_example.py]: Thanks. I've opted for 1) default arguments, as they need to be set by tests. Also for resemblance to existing samples.

**Review by kurtisvg — APPROVED** [review]

File: samples/snippets/create_lite_pubsub_export_subscription_example.py (hunk 4 above)
Comment (anguillanneuf) [reply samples/snippets/create_lite_pubsub_export_subscription_example.py:80]: Yes, this is correct.

File: samples/snippets/create_lite_pubsub_export_subscription_example.py [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+
+def create_lite_pubsub_export_subscription(
+    project_number,
+    cloud_region="us-central1",
+    zone_id="a",
+    topic_id="my-topic-id",
+    subscription_id="my-subscription-id",
+    pubsub_topic_id="destination-topic-id",
+    regional=True,
+    target_location=BacklogLocation.END,
```
Comment (anguillanneuf) [inline samples/snippets/create_lite_pubsub_export_subscription_example.py:43]: Could we match this with what the Java sample has?

```java
    // Target location within the message backlog that the subscription should be initialized to.
    SeekTarget initialLocation = SeekTarget.of(BacklogLocation.BEGINNING);
```

Comment (tmdiep) [reply samples/snippets/create_lite_pubsub_export_subscription_example.py]: Done

**Review by anguillanneuf — APPROVED** [review]

**Pull request merged**